tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/WwwAuthenticateParametersIntegrationTests.cs

```diff
@@ -60,5 +75,116 @@ public async Task CreateWwwAuthenticateResponseFromAzureResourceManagerUrlAsync(
             Assert.IsNull(authParams.Claims);
             Assert.AreEqual("invalid_token", authParams.Error);
         }
+
+        [TestMethod]
+        public async Task ExtractNonceFromWwwAuthHeadersAsync()
+        {
+            //Arrange & Act
+            //Test for nonce in WWW-Authenticate header
+            var parameterList = await WwwAuthenticateParameters.CreateFromAuthenticationResponseAsync(
+                                                         "https://testingsts.azurewebsites.net/servernonce/invalidsignature").ConfigureAwait(false);
+
+            //Assert
+            Assert.IsTrue(parameterList.Any(param => param.AuthScheme == Constants.PoPAuthHeaderPrefix));
+            Assert.IsNotNull(parameterList.Where(param => param.AuthScheme == Constants.PoPAuthHeaderPrefix).Single().PopNonce);

```

Comment:
nit: Simpler to use `enumerable.Single(condition)` instead of `enumerable.Where(condition).Single()`